ProcessLib/CompareJacobiansJacobianAssembler.cpp

```diff
@@ -0,0 +1,424 @@
+                  << '\n';
+    }
+
+    if (tol_exceeded)
+    {
+        _log_file << "# " << msg_tolerance.str() << "\n\n";
+    }
+
+    if (output)
+    {
+        dump_py(_log_file, "counter", _counter);
+        dump_py(_log_file, "num_dof", num_dof);
+        dump_py(_log_file, "abs_tol", _abs_tol);
+        dump_py(_log_file, "rel_tol", _rel_tol);
+
+        _log_file << '\n';
+
+        dump_py(_log_file, "local_x", local_x);
+        dump_py(_log_file, "local_x_dot", local_xdot);
+        dump_py(_log_file, "dxdot_dx", dxdot_dx);
+        dump_py(_log_file, "dx_dx", dx_dx);
+
+        _log_file << '\n';
+
+        dump_py(_log_file, "Jacobian_1", local_Jac1);
+        dump_py(_log_file, "Jacobian_2", local_Jac2);
+
+        _log_file << '\n';
+
+        _log_file << "# Jacobian_2 - Jacobian_1\n";
+        dump_py(_log_file, "abs_diff", abs_diff);
+        _log_file << "# Componentwise: 2 * abs_diff / (|Jacobian_1| + "
+                     "|Jacobian_2|)\n";
+        dump_py(_log_file, "rel_diff", rel_diff);
+
+        _log_file << '\n';
+
+        _log_file << "# Masks: 0 ... tolerance met, 1 ... tolerance exceeded\n";
+        dump_py(_log_file, "abs_diff_mask", abs_diff_mask);
+        dump_py(_log_file, "rel_diff_mask", rel_diff_mask);
+
+        _log_file << '\n';
+
+        dump_py(_log_file, "M_1", local_M1);
+        dump_py(_log_file, "M_2", local_M2);
+        if (fatal_error && local_M1.size() == local_M2.size())
+        {
+            dump_py(_log_file, "delta_M", local_M2 - local_M1);
+            _log_file << '\n';
+        }
+
+        dump_py(_log_file, "K_1", local_K1);
+        dump_py(_log_file, "K_2", local_K2);
+        if (fatal_error && local_K1.size() == local_K2.size())
+        {
+            dump_py(_log_file, "delta_K", local_K2 - local_K1);
+            _log_file << '\n';
+        }
+
+        dump_py(_log_file, "b_1", local_b_data);
+        dump_py(_log_file, "b_2", local_b_data2);
+        if (fatal_error && local_b1.size() == local_b2.size())
+        {
+            dump_py(_log_file, "delta_b", local_b2 - local_b1);
+            _log_file << '\n';
+        }
+
+        dump_py(_log_file, "res_1", res1);
+        dump_py(_log_file, "res_2", res2);
+        if (fatal_error)
+        {
+            dump_py(_log_file, "delta_res", res2 - res1);
+        }
+
+        _log_file << '\n';
+
+        _log_file << "### counter: " << std::to_string(_counter) << " (end)\n";
+    }
+
+    if (fatal_error)
+    {
+        OGS_FATAL("%s", msg_fatal.c_str());
+    }
+
+    if (tol_exceeded && _fail_on_error)
+    {
+        OGS_FATAL(
+            "OGS failed, because the two Jacobian implementations returned "
+            "different results.");
+    }
+}
+
+std::unique_ptr<CompareJacobiansJacobianAssembler>
+createCompareJacobiansJacobianAssembler(BaseLib::ConfigTree const& config)
+{
+    // TODO doc script corner case: Parameter could occur at different
+    // locations.
```

Comment:
note to myself. ?.

Reply:
The parameter could be at
OpenGeoSys/processes/process/jacobian_assembler/type
or at
OpenGeoSys/processes/process/jacobian_assembler/CompareJacobians/jacobian_assembler/type
in the prj file. AFAIK this is not covered by the input parameter documentation scripts.